scene/main/viewport.cpp

```diff
@@ -4123,6 +4162,12 @@ Transform2D SubViewport::get_screen_transform() const {
 	return container_transform * Viewport::get_screen_transform();
 }
 
+#ifndef _3D_DISABLED
```

Comment:
Don't do this, use xr is definately needed on subviewports. The use case here is when you add a separate viewport for rendering XR because you want to control the content displayed on the desktop. 

In this case when `use_xr` is turned off, basically the viewport becomes inactive. This is already accomplished because we trigger the size going to 0,0 as we can't obtain a window size.

I can't think of a use case where a user would re-use such a viewport for non-XR stuff but if they do, they can simply set the size they want after turning `use_xr` off.

Reply:
Ok!

I think we'll need to clear a bunch of member variables when enable XR on a SubViewport, so that when it comes out of XR it'll work right. For example, setting `size_2d_override_stretch` to false, because if the developer set it to true before enabling XR, when it comes out of XR, the `set_size_2d_override_stretch()` method won't behave correctly since it checks if the argument matches the member variable. There's a couple of these. And we'll need to add some guards to a bunch of these setters, so they don't do anything when XR is enabled.

I'll add this in my next update!